db/db_range_del_test.cc

```diff
@@ -110,6 +110,28 @@ TEST_F(DBRangeDelTest, CompactionOutputHasOnlyRangeTombstone) {
                          kSkipFIFOCompaction));
 }
 
+TEST_F(DBRangeDelTest, DBInvalidRangeCascadingProblem) {
+  DB* db;
+
+  Options options;
+  options.create_if_missing = true;
+  options.merge_operator = MergeOperators::CreateFromStringId("stringappend");
+  std::string dbname = test::PerThreadDBPath("db_write_batch_problem");
+
+  ASSERT_OK(DestroyDB(dbname, options));
+  Status s = DB::Open(options, dbname, &db);
+  ASSERT_OK(s);
+
+  WriteOptions write_options;
+
+  Status s1 = db->DeleteRange(write_options, nullptr, "Y", "Z");
+  ASSERT_OK(s1);
+  Status s2 = db->DeleteRange(write_options, nullptr, "Z", "Y");
+  ASSERT_TRUE(s2.IsInvalidArgument());
+  Status s3 = db->DeleteRange(write_options, nullptr, "A", "B");
+  ASSERT_OK(s3);  // This should work as the db is still alive.
+}
```

Comment:
Should you close/destroy the DB when the test completes or (preferably) can that be done in the destructor?